src/Libs/Task/Task.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Acquia\Console\Acsf\Libs\Task;
+
+/**
+ * Represents a task in ACSF.
+ */
+class Task {
+
+  public const TYPE_BACKUP = 'backup';
+  public const TYPE_DRUSH_CMD = 'drush_cmd';
+  public const TYPE_VERSIONS = 'versions';
+
+  /**
+   * The task id.
+   *
+   * @var string
+   */
+  private string $taskId;
+
+  /**
+   * The site id.
+   *
+   * @var string
+   */
+  private string $siteId;
+
+  /**
+   * The task type.
+   *
+   * @var string
+   */
+  private string $type;
+
+  /**
+   * Indicator whether the task is running or not.
+   *
+   * Warning: this can be outdated so it is advised to refresh the request at
+   * certain intervals.
+   *
+   * @var bool
+   */
+  private bool $isRunning;
+
+  /**
+   * Constructs a new Task object.
+   *
+   * @param string $taskId
+   *   The task id.
+   * @param string $siteId
+   *   The site id.
+   * @param string $type
+   *   The task type.
+   * @param bool $isRunning
+   *   Indication whether the task is in progress.
+   */
+  public function __construct(string $taskId, string $siteId, string $type, bool $isRunning) {
```

Comment:
Better to use snake case instead of camel case for siteId and isRunning in constructor parameters.

Reply:
oh yeah, surely I'll change it back. Looks like I forgot here, thanks!